internal/controller/ingress/domain_controller.go

```diff
@@ -142,7 +141,8 @@ func (r *DomainReconciler) create(ctx context.Context, domain *v1alpha1.Domain)
 	// errors right now, so we'll check if the domain already exists before trying to create it.
 	resp, err := r.findReservedDomainByHostname(ctx, domain.Spec.Domain)
 	if err != nil {
-		return err
+		// Set conditions before returning error
+		return r.updateStatus(ctx, domain, nil, err)
```

Comment:
this function signature of having to pass nils in some cases wasn't my favorite but was cleaner than making a stuct to pass in. I could potentially make a different `updateStatusWithError` if that would be better

Reply:
I think it is fine for now.